src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestChannelInternal.java

```diff
@@ -500,12 +459,84 @@ private void checkValidation() {
     }
   }
 
-  OpenChannelRequest.OnErrorOption getOnErrorOption() {
-    return this.onErrorOption;
-  }
-
   /** Returns underlying channel's row buffer implementation. */
   RowBuffer<T> getRowBuffer() {
     return rowBuffer;
   }
+
+  /** Returns underlying channel's attributes. */
+  public ChannelContext getChannelContext() {
+    return channelContext;
+  }
+
+  /** Channel attributes. */
+  static class ChannelContext {
```

Comment:
Could we add getter functions as well?

Reply:
I understand getters as a part of public API to get a chance of adding later some hidden details but I do not understand the value of getters for internal classes. I would not clutter the code now with getters. If this code grows and e.g. we refactor this into different packages then I would add getters. Now it is just a simple immutable struct.